pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -941,16 +942,20 @@ protected SegmentBuildDescriptor buildSegmentInternal(boolean forCommit) {
           TimeUnit.MILLISECONDS.toSeconds(waitTimeMillis));
 
       if (forCommit) {
-        File segmentTarFile = new File(dataDir, _segmentNameStr + TarGzCompressionUtils.TAR_GZ_FILE_EXTENSION);
-        try {
-          TarGzCompressionUtils.createTarGzFile(indexDir, segmentTarFile);
-        } catch (IOException e) {
-          String errorMessage =
-              String.format("Caught exception while taring index directory from: %s to: %s", indexDir, segmentTarFile);
-          _segmentLogger.error(errorMessage, e);
-          _realtimeTableDataManager
-              .addSegmentError(_segmentNameStr, new SegmentErrorInfo(now(), errorMessage, e));
-          return null;
+        File segmentTarFile = null;
+        if (!_tableConfig.getValidationConfig().isSegmentUploadToDeepStoreDisabled()) {
+          segmentTarFile = new File(dataDir, _segmentNameStr + TarGzCompressionUtils.TAR_GZ_FILE_EXTENSION);
+          try {
+            TarGzCompressionUtils.createTarGzFile(indexDir, segmentTarFile);
+          } catch (IOException e) {
+            String errorMessage =
+                String.format("Caught exception while taring index directory from: %s to: %s", indexDir, segmentTarFile);
+            _segmentLogger.error(errorMessage, e);
+            _realtimeTableDataManager.addSegmentError(_segmentNameStr, new SegmentErrorInfo(now(), errorMessage, e));
+            return null;
```

Comment:
shall this return or continue (as if deep store was disabled)

Reply:
if deepstore is disabled, the protocol will continue because it still needs to send metadata file info about the segment to controller for segment completion.

Reply:
then it shall not return?

Reply:
this behavior is the same as before and not changed (note the if condition is !deepStoreDisabled()). The idea then was then if segment tar/zip fails, the segment completion protocol just failed.

Reply:
I see. Sounds good.